api/client/catalog/categories.go

```diff
@@ -97,7 +97,21 @@ func (c *RudderDataCatalog) GetCategory(ctx context.Context, id string) (*Catego
 }
 
 func (c *RudderDataCatalog) GetCategories(ctx context.Context, options ListOptions) ([]*Category, error) {
-	return getAllResourcesWithPagination[*Category](ctx, c.client, fmt.Sprintf("v2/catalog/categories%s", options.ToQuery()))
+	cats, err := getAllResourcesWithPagination[*Category](ctx, c.client, fmt.Sprintf("v2/catalog/categories%s", options.ToQuery()))
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: External ID filterin is not working properly so we are filtering here as well
+	// until this is resovled in the backend
+	filtered := make([]*Category, 0, len(cats))
+	for _, cat := range cats {
+		if cat.ExternalID != "" {
+			filtered = append(filtered, cat)
+		}
+	}
+
```

Comment:
The API call is used both my import and load resources from remote. This fix won't work for importable categories right ?

Reply:
You are absolutely right, I should have taken the options into account for filtering